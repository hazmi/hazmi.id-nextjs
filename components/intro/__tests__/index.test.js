import React from 'react'
import renderer from 'react-test-renderer'
import Intro from '../index'

describe('With Snapshot Testing', () => {
  it('show the correct content', () => {
    const component = renderer.create(<Intro />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})