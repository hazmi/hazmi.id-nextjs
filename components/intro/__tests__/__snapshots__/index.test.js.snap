// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing show the correct content 1`] = `
<section
  className="intro"
>
  <p>
    Hello!
  </p>
  <p>
    My name is 
    <strong>
      Hazmi Abdun Nazir
    </strong>
    . I create mobile apps and websites, as a hobby and for a living.
  </p>
  <section>
    <h3>
      Contact
    </h3>
    <p>
      If you need anything, I'm on 
      <a
        aria-label="Go to Hazmi's LinkedIn"
        href="https://www.linkedin.com/in/hazmi-abdun-nazir/"
      >
        LinkedIn
      </a>
      , 
      <a
        aria-label="Go to Hazmi's Github"
        href="https://github.com/hazmi"
      >
        GitHub
      </a>
       or 
      <a
        aria-label="Go to Hazmi's Twitter"
        href="https://twitter.com/hazmi"
      >
        Twitter
      </a>
      .
    </p>
  </section>
  <section>
    <h3>
      About this site
    </h3>
    <p>
      This site is built using 
      <a
        aria-label="Go to Next.js website"
        href="https://nextjs.org/"
      >
        Next.js
      </a>
      , a 
      <a
        aria-label="Go to React website"
        href="https://reactjs.org/"
      >
        React
      </a>
       framework. The framework has export to static as well as the SSR (Server Side Rendering) options that makes the site play well with search engine. I like it.
    </p>
    <figure>
      <img
        alt="Lighthouse Audit"
        src="/static/lighthouse-audit.jpg"
      />
      <figcaption>
        <a
          aria-label="Go to Google Lighthouse website"
          href="https://developers.google.com/web/tools/lighthouse/#devtools"
        >
          Google Lighthouse
        </a>
         audit result in Chrome.
      </figcaption>
    </figure>
    <p>
      The other feature this site has is the full offline support. 
      <a
        arial-label="Go to next offline plugin page"
        href="https://github.com/hanford/next-offline"
      >
        A service worker via Google's workbox
      </a>
       combined with 
      <a
        aria-title="Go to next manifest website"
        href="https://github.com/ragingwind/next-manifest"
      >
        Web Manifest
      </a>
       makes a good 
      <a
        aria-label="PWA article on Google"
        href="https://developers.google.com/web/progressive-web-apps/"
      >
        PWA (Progressive Web App)
      </a>
       website.
    </p>
    <p>
      This means, you can do "Add to Home Screen" in iOS or Android, then can still browse the site in airplane mode.
    </p>
  </section>
  <section>
    <p>
      Do enjoy the site, may you find something useful here.
    </p>
    <p>
      Cheers!
    </p>
  </section>
</section>
`;
